contracts/nft/mainnet/NFTMainnetBridgingToPolygonFacet.sol

```diff
@@ -0,0 +1,143 @@
+    MainnetTellerNFT public constant TELLER_NFT_V2 =
+        MainnetTellerNFT(0x98Ca52786e967d1469090AdC075416948Ca004A7);
+
+    /**
+     * @notice It makes a function call to the ROOT_CHAIN_MANAGER "depositFor"
+     *  function signature, which calls our PolyTellerNFT using the CHILD_CHAIN_MANAGER
+     *  (after we map)
+     * @param tokenIds the token ids that we are depositing on ROOT_CHAIN_MANAGER
+     */
+    function __depositForV1(uint256[] memory tokenIds) internal virtual {
+        uint256[] memory amounts = new uint256[](tokenIds.length);
+        for (uint256 i; i < amounts.length; i++) {
+            amounts[i] = 1;
+        }
+        // Deposit tokens by calling the RootChainManager which then has the ERC721_PREDICATE transfer the tokens
+        Address.functionCall(
+            ROOT_CHAIN_MANAGER,
+            abi.encodeWithSignature(
+                "depositFor(address,address,bytes)",
+                msg.sender,
+                address(TELLER_NFT_V2),
+                abi.encode(tokenIds, amounts)
+            )
+        );
+    }
+
+    /**
+     * @notice It initializes our NFTBridge by approving all NFTs to the Polygon predicate contracts
+     */
+    function initNFTBridge() external {
+        TELLER_NFT_V1.setApprovalForAll(ERC721_PREDICATE, true);
+        TELLER_NFT_V2.setApprovalForAll(ERC1155_PREDICATE, true);
+    }
+
+    /**
+     * @notice Gets all of our NFTs (staked and unstaked) migrates to ERC1155 (if ERC721), then
+     *  bridges to polygon
+     * @param tokenIds The tokenIds that we are sending. First array is the ERC721, if any. Second array
+     *  are our ERC-1155 tokenIDs
+     */
+    function bridgeNFTsV1(uint256[] calldata tokenIds) external {
+        EnumerableSet.UintSet storage stakedNFTs = NFTLib.s().stakedNFTs[
+            msg.sender
+        ];
+
+        // we are creating a new array so that we can overrwrite each element
+        // with a newTokenId.
+        uint256[] memory tokenIds_ = tokenIds;
+        for (uint256 i; i < tokenIds_.length; i++) {
+            if (EnumerableSet.contains(stakedNFTs, tokenIds_[i])) {
+                NFTLib.unstake(tokenIds_[i]);
+            } else if (TELLER_NFT_V1.ownerOf(tokenIds_[i]) == msg.sender) {
+                TELLER_NFT_V1.transferFrom(
+                    msg.sender,
+                    address(this),
+                    tokenIds_[i]
+                );
+            }
+            uint256 newTokenId = TELLER_NFT_V2.convertV1TokenId(tokenIds_[i]);
+            TELLER_NFT_V1.safeTransferFrom(
+                address(this),
+                address(TELLER_NFT_V2),
+                tokenIds_[i],
+                abi.encode(newTokenId)
+            );
+            tokenIds_[i] = newTokenId;
+        }
+        __depositForV1(tokenIds_);
+    }
+
+    function bridgeNFTsV2(
+        uint256[] calldata tokenIds,
+        uint256[] calldata amounts
+    ) external {
+        // TODO: waiting for code Syed is working on
```

Comment:
just a reminder this has missing code apparently

Reply:
what is missing from this?